Source/DiabloUI/settingsmenu.cpp

```diff
@@ -142,7 +142,8 @@ bool ChangeOptionValue(OptionEntryBase *pOption, size_t listIndex)
 
 void ItemSelected(int value)
 {
-	auto &vecItem = vecDialogItems[value];
+	const auto index = static_cast<size_t>(value);

```

Comment:
Was wondering why not change the function signature, but then I realised every menu is driven by function pointers 😂 .

Another thing I noticed is the inconsistent use of `std::size_t` vs `size_t` in the current code, is the typedef in the `std` namespace is preferred for c++? I've been trying to use that where I remember.

Reply:
It was the case that `std::size_t` was preferred but in C++20 `size_t` is guaranteed to be defined in `<cstdef>` (and `uint_8` in `<cstdint>`) without the `std` prefix.

This may change yet again with C++23 stdlib modules. 😅